src/ECharts/Commands.purs

```diff
@@ -3,1115 +3,1128 @@ module ECharts.Commands where
+  F.for_ (keys ∷ Array Foreign) $ set "barWidth"
+
+setX ∷ ∀ i m. Monad m ⇒ DSL TP.DimensionI m → DSL (x ∷ I|i) m
+setX dsl = do
+  keys ← lastWithKeys ["pixels", "percents"] dsl
+  F.for_ (keys ∷ Array Foreign) $ set "x"
+
+setY ∷ ∀ i m. Monad m ⇒ DSL TP.DimensionI m → DSL (y ∷ I|i) m
+setY dsl = do
+  keys ← lastWithKeys ["pixels", "percents"] dsl
+  F.for_ (keys ∷ Array Foreign) $ set "y"
+
+setZ ∷ ∀ i m. Monad m ⇒ DSL TP.DimensionI m → DSL (z ∷ I|i) m
+setZ dsl = do
+  keys ← lastWithKeys ["pixels", "percents"] dsl
+  F.for_ (keys ∷ Array Foreign) $ set "z"
+
+coordXIx ∷ ∀ i m. Monad m ⇒ Int → DSL (x ∷ I|i) m
 coordXIx = set "x" <<< toForeign
 
-coordXValue ∷ ∀ i. String → DSL (x ∷ I|i)
+coordXValue ∷ ∀ i m. Monad m ⇒ String → DSL (x ∷ I|i) m
 coordXValue = set "x" <<< toForeign
 
-coordY ∷ ∀ i. String → DSL (y ∷ I|i)
+coordY ∷ ∀ i m. Monad m ⇒ String → DSL (y ∷ I|i) m
 coordY = set "y" <<< toForeign
 
-pixels ∷ ∀ i. Int → DSL (pixels ∷ I|i)
+pixels ∷ ∀ i m. Monad m ⇒ Int → DSL (pixels ∷ I|i) m
 pixels = set "pixels" <<< toForeign
 
-percents ∷ ∀ i. Number → DSL (percents ∷ I|i)
+percents ∷ ∀ i m. Monad m ⇒ Number → DSL (percents ∷ I|i) m
 percents = set "percents" <<< toForeign <<< (_ <> "%") <<< show
 
-setWidth ∷ ∀ i. DSL TP.DimensionI → DSL (width ∷ I|i)
-setWidth dsl =
-  F.traverse_ (set "width") $ lastWithKeys ["pixels", "percents"] dsl
+setWidth ∷ ∀ i m. Monad m ⇒ DSL TP.DimensionI m → DSL (width ∷ I|i) m
+setWidth dsl = do
+  keys ← lastWithKeys ["pixels", "percents"] dsl
+  F.for_ (keys ∷ Array Foreign) $ set "width"
 
-buildGaugeRadius ∷ ∀ i. DSL TP.DimensionI → DSL (gaugeRadius ∷ I|i)
-buildGaugeRadius dsl =
-  F.traverse_ (set "radius") $ lastWithKeys ["pixels", "percents"] dsl
+buildGaugeRadius ∷ ∀ i m. Monad m ⇒ DSL TP.DimensionI m → DSL (gaugeRadius ∷ I|i) m
+buildGaugeRadius dsl = do
+  keys ← lastWithKeys ["pixels", "percents"] dsl
+  F.for_ (keys ∷ Array Foreign) $ set "radius"
 
-buildOffsetCenter ∷ ∀ i. DSL TP.PointI → DSL (offsetCenter ∷ I|i)
-buildOffsetCenter dsl =
-  let
-    xx = get "x" dsl
-    yy = get "y" dsl
-  in
-    set "offsetCenter" $ toForeign $ Arr.catMaybes [ xx, yy ]
+buildOffsetCenter ∷ ∀ i m. Monad m ⇒ DSL TP.PointI m → DSL (offsetCenter ∷ I|i) m
+buildOffsetCenter dsl = do
+  xx ← get "x" dsl
+  yy ← get "y" dsl
+  set "offsetCenter" $ toForeign $ Arr.catMaybes [ xx, yy ]
 
-containLabel ∷ ∀ i. Boolean → DSL (containLabel ∷ I|i)
+containLabel ∷ ∀ i m. Monad m ⇒ Boolean → DSL (containLabel ∷ I|i) m
 containLabel = set "containLabel" <<< toForeign
 
-polarCoordinateSystem ∷ ∀ i. DSL (coordinateSystem ∷ I|i)
+polarCoordinateSystem ∷ ∀ i m. Monad m ⇒ DSL (coordinateSystem ∷ I|i) m
 polarCoordinateSystem = set "coordinateSystem" $ toForeign "polar"
 
-cartesianCoordinateSystem ∷ ∀ i. DSL (coordinateSystem ∷ I|i)
+cartesianCoordinateSystem ∷ ∀ i m. Monad m ⇒ DSL (coordinateSystem ∷ I|i) m
 cartesianCoordinateSystem = set "coordinateSystem" $ toForeign "cartesian2d"
 
-geoCoordina
```

Comment:
Indentation got a little messed up here.

Reply:
? I can't see your comment in pr files 0_o